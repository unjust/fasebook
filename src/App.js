import React from 'react';

import { Switch, Route, BrowserRouter } from 'react-router-dom';
import Login from './components/Login';
import ProfilePage from './components/ProfilePage';
import ProtectedRoute from './components/ProtectedRoute';

function App() {
  return (
    <BrowserRouter>
      <Switch>
        <Route path="/login" component={Login} />
        <ProtectedRoute
          path="/"
          render={(props) =>
            <ProfilePage {...props} isOwner userId={sessionStorage.getItem('userId')} />
          }
        />
      </Switch>
    </BrowserRouter>
  );
}

export default App;
