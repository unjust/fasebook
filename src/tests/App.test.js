import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import App from '../App';

describe('App component', () => {
  it('renders a BrowserRouter component', () => {
    const app = TestRenderer.create(<App />);
    const instance = app.root;
    expect(() => instance.findByType(BrowserRouter)).not.toThrowError();
  });

  it('matches snapshot', () => {
    const app = TestRenderer.create(<App />);
    const tree = app.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
