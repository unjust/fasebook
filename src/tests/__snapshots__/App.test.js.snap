// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div>
  <div
    className="card"
  >
    <h1>
      Fasebook
    </h1>
    <div
      className="margin--vert"
    >
      <label
        htmlFor="username"
      >
        Usario
      </label>
      <input
        name="username"
        type="text"
      />
    </div>
    <div
      className="margin--vert"
    >
      <label
        htmlFor="password"
      >
        Clave
      </label>
      <input
        name="password"
        type="password"
      />
    </div>
    <button
      className="button"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
